# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## My Lane: CTR / Engagement Opportunity Scoring

I chose the CTR / Engagement Opportunity Scoring lane because I want to identify pages that receive search visibility but may not receive as many clicks as expected. This lane can help prioritize pages for review by considering signals such as impressions, average search position, and click-through rate. I chose this direction because it connects measurable search data to a practical decision about which pages should be reviewed first.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

Which pages with meaningful search visibility have relatively low click-through rates and should be prioritized for review?

**Unit of analysis:** One page.

**Decision:** Decide which pages should be reviewed first for possible improvements to titles, descriptions, or search presentation.

**Output:** A ranked opportunity score or prioritized list of pages.

**Action:** An SEO or content team can review the highest-priority pages and decide whether their search presentation should be improved.

**Cost of a wrong recommendation:** A false positive could waste time reviewing a page that does not need improvement. A false negative could miss a page with a genuine opportunity to gain more clicks. Therefore, the output should support human review rather than automatically changing content.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [6]:
import pandas as pd

url = "https://raw.githubusercontent.com/Divyani117/flyrank-ml-internship-starter/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Number of pages:", len(df))
print("Median CTR:", df["ctr"].median())
print("Median impressions_90d:", df["impressions_90d"].median())
print("Median average position:", df["avg_position"].median())

Number of pages: 30000
Median CTR: 0.07
Median impressions_90d: 731.0
Median average position: 10.8


## Quick Look at the Data

The starter dataset contains 30,000 pages. The median click-through rate (CTR) is 0.07 (7%), the median number of impressions over 90 days is 731, and the median average search position is 10.8.

These numbers show that the dataset contains a substantial number of pages with measurable search visibility and engagement data. This makes the CTR / Engagement Opportunity Scoring lane worth investigating because pages with meaningful impressions but relatively low CTR may represent useful candidates for human review and prioritization.

These are descriptive observations from the starter dataset and do not prove that changing a page will increase clicks.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## What I Can and Cannot Claim

I can identify observed patterns in the dataset and rank pages based on signals such as impressions, CTR, and average search position. I can use these patterns to identify potential opportunities and support decisions about which pages should be reviewed first.

I cannot claim that low CTR is caused by a specific page problem, that changing a title or description will definitely increase clicks, or that the observed relationships prove how Google's ranking algorithm works. The results should be treated as directional evidence and decision support, not as proof of causation.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.